server/startup/migrations/v150.js

```diff
@@ -0,0 +1,14 @@
+import { Settings } from '../../../app/models';
+import { Migrations } from '../../../app/migrations/server';
+
+Migrations.add({
+	version: 150,
+	up() {
+		if (Settings.findById('AutoTranslate_GoogleAPIKey').count()) {
+			Settings.renameSetting('AutoTranslate_GoogleAPIKey', 'AutoTranslate_APIKey');
+		}
+	},
+	down() {
+		// No downgrade migrations applicable
```

Comment:
in this case a migration down applies.. you should rename the setting back from `AutoTranslate_APIKey` to `AutoTranslate_GoogleAPIKey` .. a check might be done though, to see if the new value is not for DEEPL.

Reply:
With this approach of having own settings for each provider we might not need migrations afaik. As it would mean we extend the settings for new provider.